### PR TITLE
Move to jbuilder

### PR DESCRIPTION
**Pull request by rgrinberg**
Also adds an opam file

### CODE REVIEW

File: zed.opam [hunk 1]
```diff
@@ -0,0 +1,21 @@
+opam-version: "1.2"
+maintainer: "opam-devel@lists.ocaml.org"
+authors: ["Jérémie Dimino"]
+homepage: "https://github.com/diml/zed"
+license: "BSD3"
+build: [
```
Comment (diml) [inline zed.opam]: This should probably go

File: zed.opam [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "camomile" {>= "0.8"}
+  "react"
+]
+build: [
+  ["jbuilder" "build" "--only-packages" "zed" "--root" "."
+   "-j" jobs "@install"]
+]
+dev-repo: "git://github.com/diml/zed"
+available: ocaml-version >= "4.02.3"
+install: ["ocaml" "setup.ml" "-install"]
```
Comment (diml) [inline zed.opam]: This as well

**Comment by rgrinberg** [comment]
>diml requested changes on this pull request.

Fixed.

**Comment by diml** [comment]
Merged, thanks!

**Comment by copy** [comment]
I think this broke lambda-term:


```
#=== ERROR while installing lambda-term.1.10.1 ================================#
# opam-version 1.2.2
# os           linux
# command      ocaml setup.ml -build
# path         /home/fabian/.opam/4.04.0/build/lambda-term.1.10.1
# compiler     4.04.0
# exit-code    1
# env-file     /home/fabian/.opam/4.04.0/build/lambda-term.1.10.1/lambda-term-23591-6d07ae.env
# stdout-file  /home/fabian/.opam/4.04.0/build/lambda-term.1.10.1/lambda-term-23591-6d07ae.out
# stderr-file  /home/fabian/.opam/4.04.0/build/lambda-term.1.10.1/lambda-term-23591-6d07ae.err
### stdout ###
# [...]
# /home/fabian/.opam/4.04.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I src -package lwt -package lwt.react -package lwt.unix -package str -package zed -w -40 -I examples -I src -o examples/events.cmo examples/events.ml
# /home/fabian/.opam/4.04.0/bin/ocamlfind ocamlopt -c -g -annot -bin-annot -I src -package lwt -package lwt.react -package lwt.unix -package str -package zed -w -40 -I examples -I src -o examples/events.cmx examples/events.ml
# /home/fabian/.opam/4.04.0/bin/ocamlfind ocamlopt -g -I src -linkpkg -package lwt -package lwt.react -package lwt.unix -package zed -I src -I examples src/lambda-term.cmxa examples/events.cmx -o examples/events.native
# + /home/fabian/.opam/4.04.0/bin/ocamlfind ocamlopt -g -I src -linkpkg -package lwt -package lwt.react -package lwt.unix -package zed -I src -I examples src/lambda-term.cmxa examples/events.cmx -o examples/events.native
# File "_none_", line 1:
# Error: No implementations provided for the following modules:
#          Zed_rope referenced from src/lambda-term.cmxa(LTerm_text)
#          Zed_utf8 referenced from src/lambda-term.cmxa(LTerm_text),
#            src/lambda-term.cmxa(LTerm)
# Command exited with code 2.
### stderr ###
# E: Failure("Command ''/home/fabian/.opam/4.04.0/bin/ocamlbuild' src/liblambda-term_stubs.a src/dlllambda-term_stubs.so src/lambda-term.cma src/lambda-term.cmxa src/lambda-term.a src/lambda-term.cmxs examples/events.native examples/colors.native examples/colors_256.native examples/rgb.native examples/move.native examples/hello.native examples/clock.native examples/buttons.native examples/focus.native examples/checkbuttons.native examples/radiobuttons.native examples/shell.native examples/repl.native examples/modal.native examples/read_password.native examples/read_yes_no.native examples/editor.native tools/lambda_term_actions.native -tag debug' terminated with error code 10")
```

On a 4.04 switch with this git repo pinned.

**Comment by rgrinberg** [comment]
Looks like I forgot to add `(wrapped false)` here. Will fix.

@diml In the future, perhaps we can make zed use module aliases

**Comment by diml** [comment]
Indeed. Although I barely have time to touch these projects these days, so I'm not really planning to do any breaking changes myself

**Comment by rgrinberg** [comment]
@diml How do you feel about making zed a sub package of lambda term to ease maintenance? It won't break anything for users and it's possible to even do this while preserving the git history of zed. It would be prevent issues like this and it seems like zed has no users apart from lambda-term anyway.

**Comment by diml** [comment]
I don't think it would make much difference. Especially nowadays the release workflow is quite easy with tools such as opam-publish.

Moving forward, I'm planning to do only bugfix releases of utop and its dependencies, until either there is something new or someone wants to take over

**Pull request merged**